Textinput - _key_down() method now more readable, separated deletion and insertion / action keys handling

BoyanH opened the pull request:
As we fixed issue #5469 with @emomicrowave, we noticed the _key_down() method needs some refactoring (because of repetitive self.delete_selection() calls) so we tried to make a clearer separation between deleting text when selected and other key handling.

KeyWeeUsr commented:
@BoyanH Could you please rebase the branch against master so that it contains only your commits? Thanks! :)

BoyanH commented:
@KeyWeeUsr Sorry for the mess I did, but I don't think that would be a good idea. I think I need to open a further issue.

I wanted to write some tests today, but I saw you already did, so I rebased this branch to test everything and wrote some further tests, fixed a small mistake I did.

Thing is, I found out that, on selection, when the shift is released and a cursor in the other direction is pressed, the cursor doesn't jump to the other side of the selection.

I would like to commit the test and solution (hopefully), in 30 minutes or so. Then you could look at it more easily and if you want to, I could create a cleaner branch.

KeyWeeUsr commented:
The branch contains changes that either are already included in the master branch or aren't yours, therefore it's quite a good idea to rebase the branch (`git rebase master`) :)

Re the new found selection issue, I think it should be in a separate PR together with appropriate tests, so that if somebody wants to go back in the history doesn't need to guess what happened where. This PR was about refactoring the conditions (ref the original post), therefore leave the PR as it was in the original state and further changes open in another pull request(s).

BoyanH commented:
Solves #5490

BoyanH commented:
@KeyWeeUsr thanks for the suggestions, I rebased this branch on master now and removed the further tests and fixes that don't belong here

KeyWeeUsr commented:
Thanks for PR and splitting it to parts! 🍡
